### PR TITLE
CORTX-30156 : Add component pr-deployment groovy files

### PR DESCRIPTION
**Pull request by gauravchaudhari02**
Signed-off-by: Gaurav Chaudhari <gaurav.chaudhari@seagate.com>

# Problem Statement
- PR deployment groovy files were in feature branch

# Design
-  Adding all files in `main` branch.

# Coding
   Checklist for Author
-  [x] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [ ] Unit and System Tests are added
- [ ] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [ ] Testing was performed with RPM
- [x] Jenkins pipelines used for testing
    CSM - https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Cortx-Deployment/job/Generic/job/csm-agent/390/
    Utils - https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Cortx-Deployment/job/Generic/job/py-utils/372/
    Provisioner - https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Cortx-Deployment/job/Generic/job/provisioner/211/
    Motr - https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Cortx-Deployment/job/Generic/job/motr/863/
    Hare - https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Cortx-Deployment/job/Generic/job/hare/465/
    HA - https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Cortx-Deployment/job/Generic/job/ha/81/

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [ ] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [ ] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [x] JIRA number/GitHub Issue added to PR
- [x] PR is self reviewed
- [ ] Jira and state/status is updated and JIRA is updated with PR link
- [x] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

File: jenkins/internal-ci/generic/pr/cortx_deployment/ha.groovy [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+    }
+
+    parameters {  
+        string(name: 'HA_URL', defaultValue: 'https://github.com/Seagate/cortx-ha', description: 'Repo for HA')
+        string(name: 'HA_BRANCH', defaultValue: 'main', description: 'Branch for HA')
+        string(name: 'CORTX_RE_URL', defaultValue: 'https://github.com/Seagate/cortx-re', description: 'Repo for cortx-re')
+        string(name: 'CORTX_RE_BRANCH', defaultValue: 'main', description: 'Branch for cortx-re')
+
+        choice (
+            choices: ['all', 'cortx-all' , 'cortx-rgw', 'cortx-data'],
```
Comment (shailesh-vaidya) [inline jenkins/internal-ci/generic/pr/cortx_deployment/ha.groovy:25]: Remove `cortx-all` and add `cortx-control`

File: jenkins/internal-ci/generic/pr/cortx_deployment/hare.groovy [hunk 2]
```diff
@@ -0,0 +1,300 @@
...
+
+    parameters {  
+	    string(name: 'HARE_URL', defaultValue: 'https://github.com/Seagate/cortx-hare', description: 'Repo for Hare')
+        string(name: 'HARE_BRANCH', defaultValue: 'main', description: 'Branch for Hare')     
+        string(name: 'CORTX_RE_URL', defaultValue: 'https://github.com/Seagate/cortx-re', description: 'Repo for cortx-re')
+        string(name: 'CORTX_RE_BRANCH', defaultValue: 'main', description: 'Branch for cortx-re')
+        choice(name: 'DEPLOY_BUILD_ON_NODES', choices: ["Both", "1node", "3node" ], description: '''<pre>If you select Both then build will be deploy on 1 node as well as 3 node. If you select 1 node then build will be deploy on 1 node only. If you select 3 node then build will be deploy on 3 node only. 
+</pre>''')
+        choice (
+            choices: ['all', 'cortx-all' , 'cortx-rgw', 'cortx-data'],
```
Comment (shailesh-vaidya) [inline jenkins/internal-ci/generic/pr/cortx_deployment/hare.groovy:26]: Remove `cortx-all` and add `cortx-control`

File: jenkins/internal-ci/generic/pr/cortx_deployment/motr.groovy [hunk 3]
```diff
@@ -0,0 +1,344 @@
...
+    parameters {  
+	    string(name: 'MOTR_URL', defaultValue: 'https://github.com/Seagate/cortx-motr', description: 'Repo for Motr')
+        string(name: 'MOTR_BRANCH', defaultValue: 'main', description: 'Branch for Motr')
+        choice(name: 'DEPLOY_BUILD_ON_NODES', choices: ["Both", "1node", "3node" ], description: '''<pre>If you select Both then build will be deploy on 1 node as well as 3 node. If you select 1 node then build will be deploy on 1 node only. If you select 3 node then build will be deploy on 3 node only. 
+</pre>''')
+        string(name: 'CORTX_RE_URL', defaultValue: 'https://github.com/Seagate/cortx-re', description: 'Repo for cortx-re')
+        string(name: 'CORTX_RE_BRANCH', defaultValue: 'main', description: 'Branch for cortx-re')
+
+        choice (
+            choices: ['all', 'cortx-all' , 'cortx-rgw', 'cortx-data'],
```
Comment (shailesh-vaidya) [inline jenkins/internal-ci/generic/pr/cortx_deployment/motr.groovy:27]: Remove `cortx-all` and add `cortx-control`

File: jenkins/internal-ci/generic/pr/cortx_deployment/provisioner.groovy [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+        ansiColor('xterm')  
+    }
+    parameters {  
+        string(name: 'PROVISIONER_URL', defaultValue: 'https://github.com/Seagate/cortx-prvsnr.git', description: 'Repo for PROVISIONER Agent')
+        string(name: 'PROVISIONER_BRANCH', defaultValue: 'main', description: 'Branch for PROVISIONER Agent')
+        string(name: 'CORTX_RE_URL', defaultValue: 'https://github.com/Seagate/cortx-re', description: 'Repo for cortx-re')
+        string(name: 'CORTX_RE_BRANCH', defaultValue: 'main', description: 'Branch for cortx-re')
+
+        choice (
+            choices: ['all', 'cortx-all' , 'cortx-rgw', 'cortx-data'],
```
Comment (shailesh-vaidya) [inline jenkins/internal-ci/generic/pr/cortx_deployment/provisioner.groovy:23]: Remove `cortx-all` and add `cortx-control`

File: jenkins/internal-ci/generic/pr/cortx_deployment/csm-agent.groovy [hunk 5]
```diff
@@ -0,0 +1,265 @@
...
+                                string(name: 'CORTX_RE_BRANCH', value: "${CORTX_RE_BRANCH}"),
+                                string(name: 'BUILD', value: "${CORTX_BUILD}"),
+                                string(name: 'OS', value: "${OS_VERSION}"),
+                                string(name: 'CORTX_IMAGE', value: "${CORTX_IMAGE}"),
+                                string(name: 'GITHUB_PUSH', value: "yes"),
+                                string(name: 'TAG_LATEST', value: "no"),
+                                string(name: 'DOCKER_REGISTRY', value: "cortx-docker.colo.seagate.com"),
+                                string(name: 'EMAIL_RECIPIENTS', value: "DEBUG")
+                            ]
+                        env.cortx_all_image = buildCortxAllImage.buildVariables.cortx_all_image
```
Comment (shailesh-vaidya) [inline jenkins/internal-ci/generic/pr/cortx_deployment/csm-agent.groovy:222]: Remove `cortx-all` image

**Review by shailesh-vaidya — CHANGES_REQUESTED** [review]

File: jenkins/internal-ci/generic/pr/cortx_deployment/csm-agent.groovy [hunk 6]
```diff
@@ -0,0 +1,265 @@
...
+    }
+
+    parameters {  
+        string(name: 'CSM_URL', defaultValue: 'https://github.com/Seagate/cortx-manager', description: 'Repo for CSM Agent')
+        string(name: 'CSM_BRANCH', defaultValue: 'main', description: 'Branch for CSM Agent') 
+        string(name: 'CORTX_RE_URL', defaultValue: 'https://github.com/Seagate/cortx-re', description: 'Repo for cortx-re')
+        string(name: 'CORTX_RE_BRANCH', defaultValue: 'main', description: 'Branch for cortx-re')
+
+        choice (
+            choices: ['all', 'cortx-all' , 'cortx-rgw', 'cortx-data'],
```
[the remark this answers is not in the document]
Comment (gauravchaudhari02) [reply jenkins/internal-ci/generic/pr/cortx_deployment/csm-agent.groovy]: Done

File: jenkins/internal-ci/generic/pr/cortx_deployment/ha.groovy (hunk 1 above)
Comment (gauravchaudhari02) [reply jenkins/internal-ci/generic/pr/cortx_deployment/ha.groovy]: Done

File: jenkins/internal-ci/generic/pr/cortx_deployment/motr.groovy (hunk 3 above)
Comment (gauravchaudhari02) [reply jenkins/internal-ci/generic/pr/cortx_deployment/motr.groovy]: Done

File: jenkins/internal-ci/generic/pr/cortx_deployment/provisioner.groovy (hunk 4 above)
Comment (gauravchaudhari02) [reply jenkins/internal-ci/generic/pr/cortx_deployment/provisioner.groovy]: Done

File: jenkins/internal-ci/generic/pr/cortx_deployment/csm-agent.groovy [hunk 7]
```diff
@@ -0,0 +1,265 @@
...
+                        done
+                    popd
+                    # Symlink 3rdparty repo artifacts
+                    ln -s "${THIRD_PARTY_DEPS}" "${THIRD_PARTY_LOCATION}"
+                        
+                    # Symlink python dependencies
+                    ln -s "${PYTHON_DEPS}" "${PYTHON_LIB_LOCATION}"
+                '''
+
+                // sh label: 'RPM Signing', script: '''
```
[the remark this answers is not in the document]
Comment (gauravchaudhari02) [reply jenkins/internal-ci/generic/pr/cortx_deployment/csm-agent.groovy]: Done

File: jenkins/internal-ci/generic/pr/cortx_deployment/csm-agent.groovy (hunk 5 above)
Comment (gauravchaudhari02) [reply jenkins/internal-ci/generic/pr/cortx_deployment/csm-agent.groovy]: Done

File: jenkins/internal-ci/generic/pr/cortx_deployment/hare.groovy (hunk 2 above)
Comment (gauravchaudhari02) [reply jenkins/internal-ci/generic/pr/cortx_deployment/hare.groovy]: Done

**Review by shailesh-vaidya — APPROVED** [review]

**Pull request merged**